engine/src/main/java/org/hibernate/validator/internal/cfg/context/TypeConstraintMappingContextImpl.java

```diff
@@ -75,8 +71,9 @@
 	private List<Class<?>> defaultGroupSequence;
 	private Class<? extends DefaultGroupSequenceProvider<? super C>> defaultGroupSequenceProviderClass;
 
-	TypeConstraintMappingContextImpl(DefaultConstraintMapping mapping, Class<C> beanClass) {
+	TypeConstraintMappingContextImpl(DefaultConstraintMapping mapping, Class<C> beanClass, GetterPropertySelectionStrategy getterPropertySelectionStrategy) {
 		super( mapping );
+		this.javaBeanHelper = new JavaBeanHelper( getterPropertySelectionStrategy );
```

Comment:
Just a question, not sure it's easy but could we create only one and share it?

Reply:
moved the construction of the helper to `ValidatorFactoryImpl` but it is still also created in `ConfigurationImpl`